lib/rspec/core/example_group.rb

```diff
@@ -537,9 +537,15 @@ def self.store_before_context_ivars(example_group_instance)
         end
       end
 
+      # @deprecated use `RSpec.current_scope` instead
       # Returns true if a `before(:context)` or `after(:context)`
       # hook is currently executing.
       def self.currently_executing_a_context_hook?
+        RSpec.deprecate(
+          "currently_executing_a_context_hook",
+          :replacement => "RSpec.current_scope",
+        )
+
```

Comment:
Do we need to deprecate this?

Reply:
I couldn't find any usages. Apparently, we planned to make use of it in `rspec-rails`, but didn't.
We could get rid of it in 4.0, and deprecate it in 3.x.

Reply:
This is an old (and rather ineffective) implementation that I found, trying to solve the same problem as the new implementation.

Should I do anything special or are you guys gonna take care of the difference between 3.x/4.0?

Reply:
You're totally welcome to remove this now deprecated method by sending a PR against `4-0-dev` branch once this one is merged.

Reply:
A pull request to `rspec-rails` is welcome, too.